plugin/k8s_external/external.go

```diff
@@ -90,11 +91,11 @@ func (e *External) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.Ms
 
 	switch state.QType() {
 	case dns.TypeA:
-		m.Answer = e.a(svc, state)
+		m.Answer = e.a(svc, &state)
```

Comment:
state is not a pointer anywhere in coredns, why here?

Reply:
See the change in corresponding functions : e.a(...), e.aaa(...). e.srv(...):
- the domain for the A record is usually the one of the `state`, but in case of transfer, it has to the be the one of the `svc.QName()`.
(Unless I am wrong here ... and there is another way).

So either I send state in pointer, either I change signature of those functions, either I duplicate those functions for AXFR use case.